index.d.ts

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+
+type DateInfo = {
+    day: number;
+    month: number;
+    year: number;
+}
+
+type Range = {
+    from: DateInfo | null;
+    To: DateInfo | null;
+}
+
+export interface DatePickerProps {
+    value: Range | DateInfo[] | null;
+    onChange: (newValue: Range | DateInfo[] | null) => null;
+    formatInputText?: () => string;
+    inputPlaceholder?: string;
+    inputClassName?: string;
+    renderInput?: ({ ref, onFocus, onBlur }: { ref: React.RefObject<HTMLInputElement>, onFocus: () => void, onBlur: () => void }) => React.ReactElement;

```

Comment:
@Kiarash-Z is it obligated to render an input element in this render prop?

Reply:
Yeah, there's no check for the specific element type. It's completely up to the developer. It can be any element like a `div` or `span` element.